examples/lines_bars_and_markers/lines_with_ticks_demo.py

```diff
@@ -0,0 +1,31 @@
+"""
+==============================
+Lines With A Ticked Patheffect
+==============================
+
+Ticks can be added along a line to mark one side as a barrier using
+`~matplotlib.patheffects.TickedStroke`.  You can control the angle,
+spacing, and length of the ticks.
+
+The ticks will also appear appropriately in the legend.
+
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.patheffects as PathEffects
+
+fig, ax1 = plt.subplots(figsize=(6, 6))
+p1, = ax1.plot([0, 1], [0, 1],
+               path_effects=[PathEffects.withTickedStroke(spacing=7,
+                                                          angle=135)])
+
+nx = 101
+x = np.linspace(0.0, 1.0, nx)
+y = 0.3*np.sin(x*8) + 0.4
+p2, = ax1.plot(x, y, path_effects=[PathEffects.withTickedStroke()])
+
+leg = ax1.legend([p1, p2], ["Line", "Curve"], fancybox=True,
+                 loc='upper left', edgecolor='k')
```

Comment:
Some canonical changes / simplifications:
```suggestion
from matplotlib import patheffects

fig, ax = plt.subplots(figsize=(6, 6))
ax.plot([0, 1], [0, 1], label="Line",
        path_effects=[PathEffects.withTickedStroke(spacing=7, angle=135)])

nx = 101
x = np.linspace(0.0, 1.0, nx)
y = 0.3*np.sin(x*8) + 0.4
ax.plot(x, y, label="Curve", path_effects=[PathEffects.withTickedStroke()])

ax.legend()
```